lessons/basics/modules.md

```diff
@@ -249,8 +288,44 @@ defmacro __using__(_) do
 end
 ```
 
-The `Ecto.Migration.__using__/1` macro includes an import call so that when you `use Ecto.Migration` you also `import Ecto.Migration`. It also sets up a module property which we will assume controls Ecto’s behavior.
+The `Ecto.Migration.__using__/1` macro includes an import call so that when you `use Ecto.Migration` you also `import Ecto.Migration`.
+It also sets up a module property which we will assume controls Ecto’s behavior.
+
+We can also override a given function that is given to use with `__using__/1`.
+This is done by using `defoverridable`.
+Lets implement this:
+
+```elixir
+iex> defmodule Foo do
+...>   defmacro __using__(_) do
+...>     quote do
+...>       def bar do
+...>         2
+...>       end
+...>       defoverridable [bar: 0]
+...>     end
+...>   end
+...> end
+{:module, Foo,
+ <<. . .>>, {:__using__, 1}}
+iex> defmodule Baz do
+...>   use Foo
+...>   def bar do
+...>     super + 2
```

Comment:
I'm gonna go through and edit RE suggestions and such later tonight or tomorrow at the airport and push everything up.